packages/backend/src/nest/connections-manager/connections-manager.service.ts

```diff
@@ -260,144 +311,205 @@ export class ConnectionsManagerService extends EventEmitter implements OnModuleI
       return
     }
 
+    // TODO: Should we save this network info in LevelDB at this point?
     return network
   }
 
-  private async generatePSK() {
-    const pskBase64 = Libp2pService.generateLibp2pPSK().psk
-    await this.localDbService.put(LocalDBKeys.PSK, pskBase64)
-    this.logger('Generated Libp2p PSK')
-    this.serverIoProvider.io.emit(SocketActionTypes.LIBP2P_PSK_STORED, { psk: pskBase64 })
-  }
-
-  public async createCommunity(payload: InitCommunityPayload) {
+  public async createCommunity(payload: InitCommunityPayload): Promise<Community | undefined> {
     this.logger('Creating community: peers:', payload.peers)
 
-    await this.generatePSK()
+    if (!payload.CA || !payload.rootCa) {
+      this.logger.error('CA and rootCa are required to create community')
+      return
+    }
+
+    if (!payload.ownerCsr) {
+      this.logger.error('ownerCsr is required to create community')
+      return
+    }
+
+    const psk = Libp2pService.generateLibp2pPSK().psk
+    let ownerCertResult: SavedOwnerCertificatePayload
+
+    try {
+      this.emit(SocketActionTypes.CONNECTION_PROCESS_INFO, ConnectionProcessInfo.REGISTERING_OWNER_CERTIFICATE)
+      ownerCertResult = await this.registrationService.registerOwnerCertificate({
+        communityId: payload.id,
+        userCsr: payload.ownerCsr,
+        permsData: {
+          certificate: payload.CA.rootCertString,
+          privKey: payload.CA.rootKeyString,
+        },
+      })
+    } catch (e) {
+      this.logger.error('Failed to register owner certificate')
+      return
+    }
```

Comment:
We register the owner certificate from createCommunity since that's how it was done before in sagas. We previously called REGISTER_OWNER_CERTIFICATE and then when that completed, we called CREATED_COMMUNITY. Now we do it all together. However, we may want to decouple the owner cert from the community in future work.